tasmota/tasmota_xdrv_driver/xdrv_02_9_mqtt.ino

```diff
@@ -737,6 +737,21 @@ void MqttPublishPayload(const char* topic, const char* payload) {
   MqttPublishPayload(topic, payload, 0, false);
 }
 
+void MqttPublish(const char* topic, bool retained, bool binary) {
+  // Publish <topic> default ResponseData string with optional retained
+  char* response_data;
+  uint32_t binary_length = 0;
+  response_data = ResponseData();
```

Comment:
I'm not sure this is valid, since ResponseData() is `(char*)TasmotaGlobal.mqtt_data.c_str();`

I suppose this conversion shouldn't be used unless binary conversion is explicitly needed

Reply:
I'm not sure I understand what you mean. When creating a rule a binary payload cannot be specified so the character string will just contain printable characters.